### PR TITLE
Zoo clock

### PR DESCRIPTION
**Pull request by bliminate**
Added the zoo clock object, and made the zookeeper observe the zoo clock. The zookeeper now performs actions at set times of the day.

I need to add functionality to have the zookeeper arrive and leave each day at a set time too.

### CODE REVIEW

**Comment by bliminate** [comment]
I added the functionality to have the zookeeper arrive and leave based on the clock - and to have the clock run for a number of days - this removed the for loop in the main zoo.

I would like to add the zooAnouncer making an anouncement each day for when the zoo opens and closes after the zoo anouncer code gets merged into master.

File: src/observer/ClockSubject.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+   public void removeObserver(iObserver o) {
+      if(observers.contains(o)){
+         observers.remove(observers.indexOf(o));
+      }
+   }
+
+   @Override
+   public void notifyObservers() {
+      for ( iObserver o:observers ) {
+         o.update(this);
```
Comment (watakandai) [inline src/observer/ClockSubject.java:31]: Are you not sending only the variables that changed? You are just notifying that there was a change and the observer access the clock's attributes directly by holding clock as its attribute? Which implementation do you think is better? 
1. Notify changes and access via its attribute
2. Publish variables that only changed and not holding clock class as its attribute?

Comment (watakandai) [reply src/observer/ClockSubject.java:31]: Second method doesn't have to parse clock as an argument in the observer constructor. It only has to be registered to the publishers which is easier, but it really depends on how much we want to implement Observer / Publisher.

File: src/employees/ZooKeeper.java [hunk 2]
```diff
@@ -1,23 +1,41 @@
...
+import clock.Clock;
+import observer.*;
 import nameGenerator.nameGenerator;
 
 import java.util.List;
 
-public class ZooKeeper extends ZooEmployee {
+public class ZooKeeper extends ZooEmployee implements iObserver {
 
    public ZooKeeper() {
```
Comment (watakandai) [inline src/employees/ZooKeeper.java:12]: `responsibleAnimals` is null if we use the default constructor. How do you deal with this kind of problem in Java?

Comment (watakandai) [reply src/employees/ZooKeeper.java:12]: `Clock c` is also null. How can we be sure that they are all set?

File: src/test/test_src/employees/ZooKeeperTest.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import clock.*;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ZooKeeperTest {
+
+   @Test
+   void performTasks_Wakes_Animals_At_0800() {
```
Comment (watakandai) [inline src/test/test_src/employees/ZooKeeperTest.java:12]: nit: do you guys use camel cases or use underscore `_` to connect words?

**Pull request merged**